WordPress/src/main/java/org/wordpress/android/ui/prefs/language/LocalePickerAdapter.kt

```diff
@@ -0,0 +1,41 @@
+package org.wordpress.android.ui.prefs.language
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import org.wordpress.android.ui.prefs.language.LocalePickerListItem.LocalePickerListViewType.LOCALE
+import org.wordpress.android.ui.prefs.language.LocalePickerListItem.LocaleRow
+
+class LocalePickerAdapter : ListAdapter<LocalePickerListItem, LocalePickerListViewHolder<*>>(DIFF_CALLBACK) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): LocalePickerListViewHolder<*> {
+        return when (viewType) {
+            LOCALE.ordinal -> LocalePickerListItemViewHolder(parent)
+            else -> throw IllegalArgumentException("Unexpected view holder in LocalePickerAdapter")
+        }
+    }
+
+    override fun onBindViewHolder(holder: LocalePickerListViewHolder<*>, position: Int) {
+        val item = getItem(position)
+        when (holder) {
+            is LocalePickerListItemViewHolder -> holder.bind(item as LocaleRow)
+        }
+    }
+
+    override fun getItemViewType(position: Int): Int {
+        return getItem(position)!!.type.ordinal
```

Comment:
probably, not required given it's only one view type.

Reply:
I'm planning to add a sub-header for preferred and all locales during next hack week, so I left this in :)